README.rst

```diff
@@ -0,0 +1,318 @@
+found `here <https://conda.io/docs/>`__. Installing ``oommf``
+`package <https://anaconda.org/conda-forge/oommf>`__ using conda can be
+done by running
+
+::
+
+   conda install -c conda-forge oommf
+
+Support
+-------
+
+For support on OOMMF itself, please refer to OOMMF’s `web
+page <https://math.nist.gov/oommf/oommf.html>`__. However, if you have
+any difficulties or problems in using any of the features we provide in
+this repository, you are welcome to raise an issue in our
+`ubermag/help <https://github.com/ubermag/help>`__ repository.
+
+License
+-------
+
+The License for the OOMMF code can be found `here <oommf/LICENSE>`__. If
+you use any of the DMI extensions (that are not part of this
+repository), please refer to the `3-Clause BSD
+License <https://opensource.org/licenses/BSD-3-Clause>`__ and licenses
+in corresponding repositories
+(`Cnv <https://github.com/joommf/oommf-extension-dmi-cnv>`__,
+`T(O) <https://github.com/joommf/oommf-extension-dmi-t>`__,
+`D2d <https://github.com/joommf/oommf-extension-dmi-d2d>`__, and
+`magnetoelastic <https://github.com/yuyahagi/oommf-mel>`__).
+
+How to cite
+-----------
+
+Information on how to cite the usage of OOMMF can be found
+`here <https://math.nist.gov/oommf/oommf_cites.html>`__.
+
+If you use any of the DMI extensions we provide in your research, please
+refer to the “How to cite” section in the corresponding repositories for
+the particular crystallographic class
+(`Cnv <https://github.com/joommf/oommf-extension-dmi-cnv>`__,
+`T(O) <https://github.com/joommf/oommf-extension-dmi-t>`__, or
+`D2d <https://github.com/joommf/oommf-extension-dmi-d2d>`__).
+
+Acknowledgements
+----------------
+
+The repository which holds the OOMMF source code was developed as a part
+of `OpenDreamKit <http://opendreamkit.org/>`__ – Horizon 2020 European
+Research Infrastructure project (676541) and the `EPSRC Programme grant
+on Skyrmionics (EP/N032128/1) <https://www.skyrmions.ac.uk/>`__. Further
+thanks go to the University of Southampton and the Max Planck Institute
+for the Structure and Dynamics of Matter.
+
+Information for Developers
+==========================
+
+This repository is providing a git repository of oommf source files and
+additional extensions (see above for details).
+
+Updating to new OOMMF releases, or including new extensions
+-----------------------------------------------------------
+
+If a new OOMMF release (from NIST) should be included here, one should
+use the targets in the `Makefile-update-oommf <Makefile-update-oommf>`__
+makefile (see documentation there). The file will need manual updating
+(for the new version), before being used. Once it has run through, the
+new files need to be committed to the repository. This README might need
+updating (where specific version numbers are mentioned).
+
+We have a github workflow to run through the targets in this
+`Makefile-update-oommf <Makefile-update-oommf>`__ makefile periodically:
```

Comment:
Good point. I have tried to explain this, and extended that section a little as well to describe all required steps for creating a new release.

Reply:
The additional explanation/summary of the steps required to update is very good. Do I understand it correct that the targets in the Makefile will produce some zipped/tarred files that should not be committed? If so can we safely automatically delete these in the Makefile?

Reply:
Hm, not quite. The `clone-log-and-extract-src.py` program creates some zip files:
<pre>
oommf/app/oxs/local/oommf-extension-dmi-d2d/provenance-oommf-extension-dmi-d2d-HEAD.zip
oommf/app/oxs/local/oommf-extension-dmi-cnv/provenance-oommf-extension-dmi-cnv-HEAD.zip
oommf/app/oxs/local/oommf-mel/provenance-oommf-mel-HEAD.zip
oommf/app/oxs/local/oommf-extension-dmi-t/provenance-oommf-extension-dmi-t-HEAD.zip
</pre>
which provide a snapshot of the HEAD of the contributed repositories at the time where we have cloned them. These zip files should be committed.

I wanted to include them, because sometimes there are examples or documentation in the extension repositories: if they are organised well, these are not together with the actual c++ source, and thus we would not include them in our oommf release. (But I'd like to include them in case the original github repo is renamed / disappears etc: then it would be awkward to only have the source but not the documentation/examples/tests etc that came with it.)

Reply:
Okay, I see. Can we rewrite point `3.`? (Maybe just remove the part about unzipped/untarred if we can just commit all (new) files.)

Reply:
Fair point. I have updated this, and also extended the explanation for point 1 and 2 (to explain where the files come from).

Committing should probably still be a manual process: 

- if you add a new extension (for example), it needs to be added to the repository.

- committing should happen after the files have been fetched, but before they are (for testing purposes) compiled.

I have extended the explanations further to explain this.

Reply:
The extended explanation is good. Reading this, and in particular the section mentioning to repeat until the tests pass, I was wondering whether we should also add tests for the additional extensions (basically running (some of) the examples provided alongside the extensions) to ensure compatibility with updates in OOMMF.

Reply:
Yes, that would be good. But let's open a new issue  (and later maybe a PR) for this.